src/app/features/data-import/data-import/data-import.component.ts

```diff
@@ -9,10 +12,61 @@ import { DataImportService } from "../data-import.service";
 @Injectable({
   providedIn: "root",
 })
-export class DataImportComponent {
-  constructor(private dataImportService: DataImportService) {}
+export class DataImportComponent implements OnInit{
+  @Input() firstFormGroup: FormGroup;
+  @Input() secondFormGroup: FormGroup;
 
-  importCsvFile(file: Blob): void {
-    this.dataImportService.handleCsvImport(file);
+  csvFile: Blob = undefined;
+
+  constructor(
+    private dataImportService: DataImportService,
+    private _formBuilder: FormBuilder,
+    private dynamicEntityService: DynamicEntityService) {}
+
+    ngOnInit() {
+      this.firstFormGroup = this._formBuilder.group({
+        firstCtrl: ['', Validators.required],
+      });
+      this.secondFormGroup = this._formBuilder.group({
+        secondCtrl: ['', Validators.required],
+      });
+    }
+
+    getEntitiesMap(): Map<string, EntityConstructor<Entity>> {
+      return this.dynamicEntityService.EntityMap;
+    }
+  
+    get hasValidFile(): boolean {
+      return this.csvFile !== undefined;
+    }
+
+    entitySelectionChanged(): void {
+      // whenver the selection changes, the file can't be valid (if there was one)
```

Comment:
Currently this is checked by the file validation, but there is no message to the user yet, except the file is not used (silent fail).

Reply:
Makes sense, boils down to the same thing: The file should be parsed after uploading it, and the parsed result kept instead of the raw file. That way one could check those things and react properly to it and set the information in the importMetaData object